src/main/scala/it/pagopa/interop/authorizationprocess/common/AuthorizationUtils.scala

```diff
@@ -0,0 +1,35 @@
+package it.pagopa.interop.authorizationprocess.common
+
+import cats.implicits._
+import it.pagopa.interop.commons.utils.AkkaUtils.getOrganizationIdFutureUUID
+import it.pagopa.interop.authorizationprocess.error.AuthorizationProcessErrors._
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+import it.pagopa.interop.authorizationprocess.service.AuthorizationManagementService
+
+object AuthorizationUtils {
+
+  def assertIsConsumer(clientId: UUID)(
+    authorizationManagementService: AuthorizationManagementService
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Unit] =
+    for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      _              <- authorizationManagementService
+        .getClient(clientId)(contexts)
+        .ensureOr(client => OrganizationNotAllowedOnClient(clientId.toString, client.consumerId))(
+          _.consumerId == organizationId
+        )
+    } yield ()
+
+  def assertIsConsumerOfRequestedPurpose(purposeId: UUID, consumerId: UUID)(implicit
```

Comment:
another minor suggestion
```suggestion
  def assertIsPurposeConsumer(purposeId: UUID, purposeConsumerId: UUID)(implicit
```